bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirCommandType.java

```diff
@@ -0,0 +1,1024 @@
+
+        ComfoAirCommandType commandType = ComfoAirCommandType.getCommandTypeByKey(key);
+        if (commandType != null) {
+            if (commandType.read_reply_command != 0) {
+                int replyCmd = commandType.read_reply_command;
+
+                ComfoAirCommand command = new ComfoAirCommand(key);
+                commands.put(replyCmd, command);
+            }
+
+            if (commandType.change_affected != null) {
+                for (String affectedKey : commandType.change_affected) {
+                    // refresh affected event keys only when they are used
+                    if (!usedKeys.contains(affectedKey)) {
+                        continue;
+                    }
+
+                    ComfoAirCommandType affectedCommandType = ComfoAirCommandType.getCommandTypeByKey(affectedKey);
+
+                    if (affectedCommandType != null) {
+                        if (affectedCommandType.read_reply_command == 0) {
+                            continue;
+                        }
+                        commands = modifiedCommandCollection(commands, affectedCommandType);
+                    }
+                }
+            }
+        }
+        return commands.values();
+    }
+
+    /**
+     * Get all commands which receive informations to update items.
+     *
+     * @return all ComfoAirCommand's identified by keys
+     */
+    public static Collection<ComfoAirCommand> getReadCommandsByEventTypes(List<String> keys) {
+
+        Map<Integer, ComfoAirCommand> commands = new HashMap<>();
+        for (ComfoAirCommandType entry : values()) {
+            if (!keys.contains(entry.key)) {
+                continue;
+            }
+            if (entry.read_reply_command == 0) {
+                continue;
+            }
+            commands = modifiedCommandCollection(commands, entry);
+        }
+        return commands.values();
+    }
+
+    /**
+     * Get commandtypes which matches the replyCmd.
+     *
+     * @param replyCmd
+     *            reply command byte value
+     * @return ComfoAirCommandType identified by replyCmd
+     */
+    public static List<ComfoAirCommandType> getCommandTypesByReplyCmd(int replyCmd) {
+        List<ComfoAirCommandType> commands = new ArrayList<>();
+        for (ComfoAirCommandType entry : values()) {
+            if (entry.read_reply_command != replyCmd) {
+                continue;
+            }
+            commands.add(entry);
+        }
+        return commands;
+    }
+
+    /**
+     * Get a specific command.
+     *
+     * @param key
+     *            command key
+     * @return ComfoAirCommandType identified by key
+     */
+    public static @Nullable ComfoAirCommandType getCommandTypeByKey(String key) {
+        for (ComfoAirCommandType entry : values()) {
+            if (entry.key.equals(key)) {
+                return entry;
+            }
+        }
+        return null;
+    }
+
+    @SuppressWarnings("null")
```

Comment:
this would create a warning that `commands.get(replyCmd)` could never be `null` and the null-check not necessary, which is actually wrong if `commands` does not include `replyCmd`.
See also https://community.openhab.org/t/nonnullbydefault-and-hashmap/79845